stdlib/asm/collections/smt.masm

```diff
@@ -1,232 +1,351 @@
-use.std::mem
+use.std::crypto::hashes::rpo
 use.std::collections::sorted_array
+use.std::math::u64
+use.std::mem
 use.std::sys
+use.std::word
 
 # Constant value for the depth at which leaves sit
 const.LEAF_DEPTH=64
 
 # Each key-value pair is two words, and a leaf can hold up to 1024 pairs.
 const.MAX_LEAF_SIZE=8192
 
-# SET
+# EXPORTS
 # =================================================================================================
 
-#! Inserts or removes a value associated with the given key. The leaf to which we're inserting is
-#! guaranteed to be empty.
+#! Returns the value located under the specified key in the Sparse Merkle Tree defined by the
+#! specified root.
+#!
+#! If no values had been previously inserted under the specified key, an empty word (i.e.,
+#! [ZERO; 4]) is returned.
 #!
 #! Inputs:
-#!   Operand stack: [V, K, R, ...]
+#!   Operand stack: [K, R, ...]
 #!
 #! Outputs:
-#!   Operand stack: [V_old, R_new, ...]
+#!   Operand stack: [V, R, ...]
+#!
+#! Fails if the tree with the specified root does not exits in the VM's advice provider.
 #!
 #! Cycles
-#!   Insert empty value: X cycles
-#!   Insert non-empty value: X cycles
-proc.set_empty_leaf
-    # Check if we're inserting the empty value (X cycles)
+#!   Leaf empty: 66 cycles
+#!   Leaf single: 115 cycles
+#!   Leaf multiple, non-empty value: 189 + 3 * pair_count
+#!   Leaf multiple, empty value: 186 + 3 * pair_count
+export.get
```

Comment:
TODO: Consider mentioning this handles the full multi-leaf state machine (empty ↔ single ↔ multi) for clarity in the top-level export documentation.